error: invalid conversion from ‘const xmlError*’

Hi all.

`libsbml-5.20.2` is not compiling against `libxml-2.12.1`:

```
cd /builddir/build/BUILD/libsbml-5.20.2/build/src && /usr/bin/g++ -DLIBSBML_COMPILED_IN_SRC -DLINUX -DUSE_BZ2 -DUSE_COMP -DUSE_FBC=1 -DUSE_GROUPS -DUSE_L3V2EXTENDEDMATH -DUSE_LAYOUT -DUSE_LIBXML -DUSE_MULTI -DUSE_QUAL -DUSE_RENDER -DUSE_ZLIB -I/builddir/build/BUILD/libsbml-5.20.2/build/src/src/sbml -I/builddir/build/BUILD/libsbml-5.20.2/build/src/src -I/builddir/build/BUILD/libsbml-5.20.2/build/src -I/builddir/build/BUILD/libsbml-5.20.2/src -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/comp/common -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/comp/extension -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/comp/sbml -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/comp/util -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/comp/validator -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/comp/validator/constraints -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/fbc/common -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/fbc/extension -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/fbc/sbml -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/fbc/util -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/fbc/validator -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/fbc/validator/constraints -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/groups/common -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/groups/extension -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/groups/sbml -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/groups/validator -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/groups/validator/constraints -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/l3v2extendedmath/common -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/l3v2extendedmath/extension -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/l3v2extendedmath/sbml -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/l3v2extendedmath/validator -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/l3v2extendedmath/validator/constraints -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/multi/common -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/multi/extension -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/multi/sbml -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/multi/validator -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/multi/validator/constraints -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/qual/common -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/qual/extension -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/qual/sbml -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/qual/validator -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/qual/validator/constraints -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/render/common -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/render/extension -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/render/sbml -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/render/util -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/render/validator -I/builddir/build/BUILD/libsbml-5.20.2/src/sbml/packages/render/validator/constraints -isystem /usr/include/libxml2 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64   -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fno-strict-aliasing -fPIC -DNDEBUG -MD -MT src/CMakeFiles/sbml-static.dir/sbml/xml/LibXMLParser.cpp.o -MF CMakeFiles/sbml-static.dir/sbml/xml/LibXMLParser.cpp.o.d -o CMakeFiles/sbml-static.dir/sbml/xml/LibXMLParser.cpp.o -c /builddir/build/BUILD/libsbml-5.20.2/src/sbml/xml/LibXMLParser.cpp
make[2]: Leaving directory '/builddir/build/BUILD/libsbml-5.20.2/build'
/builddir/build/BUILD/libsbml-5.20.2/src/sbml/xml/LibXMLParser.cpp: In member function ‘virtual bool LibXMLParser::parseNext()’:
/builddir/build/BUILD/libsbml-5.20.2/src/sbml/xml/LibXMLParser.cpp:427:46: error: invalid conversion from ‘const xmlError*’ {aka ‘const _xmlError*’} to ‘xmlErrorPtr’ {aka ‘_xmlError*’} [-fpermissive]
  427 |     xmlErrorPtr libxmlError = xmlGetLastError();
      |                               ~~~~~~~~~~~~~~~^~
      |                                              |
      |                                              const xmlError* {aka const _xmlError*}
make[2]: *** [src/CMakeFiles/sbml-static.dir/build.make:6155: src/CMakeFiles/sbml-static.dir/sbml/xml/LibXMLParser.cpp.o] Error 1
```

--- a/src/sbml/xml/LibXMLParser.cpp
+++ b/src/sbml/xml/LibXMLParser.cpp
@@ -424,7 +424,7 @@ LibXMLParser::parseNext ()
 
   if ( xmlParseChunk(mParser, mBuffer, bytes, done) )
   {
-    xmlErrorPtr libxmlError = xmlGetLastError();
+    const xmlError* libxmlError = xmlGetLastError();
 
     // I tried reporting the message from libXML that's available in
     // libxmlError->message, but the thing is bogus: it will say things

